llvm-passes/ObfuscateVar/ObfuscateVar.cpp

```diff
@@ -0,0 +1,365 @@
+                splitVariable(V,Inst);
+                //dbgs() << *V << " is now splited\n";
+                
+              }
+
+            }//isValidCandidateOperand
+          }//for
+
+          if(BinaryOperator *Binop = dyn_cast<BinaryOperator>(&Inst)) {
+
+            //Get the operator's operands
+            Value *op0 = parseOperand(Binop->getOperand(0));
+            Value *op1 = parseOperand(Binop->getOperand(1));
+           
+            if(!(isSplited(op0) && isSplited(op1))){//Check if the two operands are splited. It should be ok due to the previous loop
+              dbgs() << "Error : operands aren't splited !\n";
+              break;
+            }
+           
+            switch(Binop->getOpcode()){
+              
+            case Instruction::Add :{//Split add instruction
+
+              dbgs() << "Split ADD instruction\n";
+              
+              IRBuilder<> Builder(&Inst);
+              Type *IntermediaryType = IntegerType::get(Inst.getParent()->getContext(),sizeof(typeInter)*8);//32bits
+    
+              Constant *C10 = ConstantInt::get(IntermediaryType,10,false);
+              
+              // Get X_A and X_B from the first operand
+              typeMap::iterator it=varsRegister.find(op0); 
+              
+              Value *op0_A = Builder.CreateLoad(it->second.first,isVolatile);
+              Value *op0_B = Builder.CreateLoad(it->second.second,isVolatile);
+
+              // Get Y_A and Y_B from the second operand
+              it=varsRegister.find(op1);
+              Value *op1_A = Builder.CreateLoad(it->second.first,isVolatile);
+              Value *op1_B = Builder.CreateLoad(it->second.second,isVolatile);
+
+              
+              Value* AddA0A1 = Builder.CreateAdd(op0_A,op1_A);//X_A+Y_A
+              Value* AddB0B1 = Builder.CreateAdd(op0_B,op1_B);//X_B+Y_B
+              
+              Value* RemA0A1 = Builder.CreateURem(AddA0A1,C10,"Z_A");//X_A+Y_A mod 10 => Z_A
+
+              Value* MulB0B1 = Builder.CreateMul(AddB0B1,C10);//10*(X_B+Y_B)
+              
+              Value* AddAB = Builder.CreateAdd(MulB0B1,AddA0A1);//10*(X_B+Y_B)+[X_A+Y_A]
+
+              Value* DivAB = Builder.CreateUDiv(AddAB,C10,"Z_B"); //{ 10*(X_B+Y_B)+[X_A+Y_A] } div 10 => Z_B
+
+              //Allocate two register to store the splited result
+              Value* alloResultA = Builder.CreateAlloca(IntermediaryType,nullptr,"a_res");
+              Value* alloResultB = Builder.CreateAlloca(IntermediaryType,nullptr,"b_res");
+
+              Builder.CreateStore(RemA0A1,alloResultA,isVolatile);
+              Builder.CreateStore(DivAB,alloResultB,isVolatile);
+ 
+              Value* LoadResultA = Builder.CreateLoad(alloResultA,isVolatile);
+
+              // the key to access to the variables Z_A and Z_B from the ValueMap is Z_A.
+              // It's a convention and we could choose Z_B
+              varsRegister[parseOperand(LoadResultA)] = std::make_pair(alloResultA,alloResultB);
```

Comment:
varsRegister could be a llvm::ValueMap that maps unsplitted value to splitted values.